### PR TITLE
Enable STM32WB ADC

### PR DESCRIPTION
**Pull request by rickconrey**

### CODE REVIEW

File: drivers/adc/adc_stm32.c [hunk 1]
```diff
@@ -513,9 +525,10 @@ static int adc_stm32_init(struct device *dev)
 	defined(CONFIG_SOC_SERIES_STM32L0X)
 	LL_ADC_SetClock(adc, LL_ADC_CLOCK_SYNC_PCLK_DIV4);
 #elif defined(CONFIG_SOC_SERIES_STM32F3X) || \
-	defined(CONFIG_SOC_SERIES_STM32L4X)
-	LL_ADC_SetCommonClock(__LL_ADC_COMMON_INSTANCE(),
-			LL_ADC_CLOCK_SYNC_PCLK_DIV4);
+	defined(CONFIG_SOC_SERIES_STM32L4X) || \
+	defined(CONFIG_SOC_SERIES_STM32WBX)
+	LL_ADC_SetCommonClock(__LL_ADC_COMMON_INSTANCE(adc),
+			LL_ADC_CLOCK_SYNC_PCLK_DIV2);
```
Comment (erwango) [inline drivers/adc/adc_stm32.c]: You're changing argument of the function. This change impacts other series.
If this is a bug you want to fix, please submit as a separate PR and explain your change, so people can have possibility to review  and test.

File: drivers/adc/adc_stm32.c [hunk 2]
```diff
@@ -539,13 +553,15 @@ static int adc_stm32_init(struct device *dev)
 	 * These two series STM32 has one internal voltage reference source
 	 * to be enabled.
 	 */
-	LL_ADC_SetCommonPathInternalCh(ADC, LL_ADC_PATH_INTERNAL_VREFINT);
+	LL_ADC_SetCommonPathInternalCh(__LL_ADC_COMMON_INSTANCE(adc),
```
Comment (erwango) [inline drivers/adc/adc_stm32.c]: Same here, please submit at least in a different commit explaining this change.

File: drivers/adc/adc_stm32.c [hunk 3]
```diff
@@ -558,17 +574,18 @@ static int adc_stm32_init(struct device *dev)
 	}
 
 	wait_cycles = CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC / adc_rate *
-		      LL_ADC_DELAY_CALIB_ENABLE_ADC_CYCLES;
+			  LL_ADC_DELAY_CALIB_ENABLE_ADC_CYCLES;
```
Comment (erwango) [inline drivers/adc/adc_stm32.c]: Remove this change

File: drivers/adc/adc_stm32.c [hunk 4]
```diff
@@ -624,14 +645,14 @@ static struct adc_stm32_data adc_stm32_data_##index = {			\
 };									\
 									\
 DEVICE_AND_API_INIT(adc_##index, DT_ADC_##index##_NAME, &adc_stm32_init,\
-		    &adc_stm32_data_##index, &adc_stm32_cfg_##index,	\
```
Comment (erwango) [inline drivers/adc/adc_stm32.c]: Remove this change

File: drivers/adc/adc_stm32.c [hunk 5]
```diff
@@ -624,14 +645,14 @@ static struct adc_stm32_data adc_stm32_data_##index = {			\
...
-		    POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,	\
-		    &api_stm32_driver_api);				\
+			&adc_stm32_data_##index, &adc_stm32_cfg_##index,	\
+			POST_KERNEL, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,	\
+			&api_stm32_driver_api);				\
 									\
 static void adc_stm32_cfg_func_##index(void)				\
 {									\
-	IRQ_CONNECT(DT_ADC_##index##_IRQ, DT_ADC_##index##_IRQ_PRI,	\
```
Comment (erwango) [inline drivers/adc/adc_stm32.c]: Remove this change

File: soc/arm/st_stm32/stm32wb/dts_fixup.h [hunk 6]
```diff
@@ -83,24 +83,31 @@
 #define DT_FLASH_DEV_BASE_ADDRESS		DT_ST_STM32WB_FLASH_CONTROLLER_58004000_BASE_ADDRESS
 #define DT_FLASH_DEV_NAME			DT_ST_STM32WB_FLASH_CONTROLLER_58004000_LABEL
 
-#define DT_I2C_1_BASE_ADDRESS                   DT_ST_STM32_I2C_V2_40005400_BASE_ADDRESS
```
Comment (erwango) [inline soc/arm/st_stm32/stm32wb/dts_fixup.h]: Limit this PR to ADC related changes

File: soc/arm/st_stm32/stm32wb/Kconfig.defconfig.series [hunk 7]
```diff
@@ -32,4 +32,11 @@ config I2C_STM32_V2
 
 endif # I2C_STM32
 
+if ADC
```
Comment (erwango) [inline soc/arm/st_stm32/stm32wb/Kconfig.defconfig.series]: Already done in `soc/arm/st_stm32/common/Kconfig.defconfig.series`.
You can remove this  change

File: tests/drivers/adc/adc_api/src/test_adc.c [hunk 8]
```diff
@@ -127,7 +127,8 @@
 	defined(CONFIG_BOARD_NUCLEO_F207ZG) || \
 	defined(CONFIG_BOARD_NUCLEO_F401RE) || \
 	defined(CONFIG_BOARD_NUCLEO_F746ZG) || \
-	defined(CONFIG_BOARD_NUCLEO_L073RZ)
+	defined(CONFIG_BOARD_NUCLEO_L073RZ) || \
+    defined(CONFIG_BOARD_NUCLEO_WB55RG)
```
Comment (erwango) [inline tests/drivers/adc/adc_api/src/test_adc.c]: Please align this line.

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for this contribution. Change looks globally good.

Please separate the changes that impacts other series, as well as formatting.
Also, fyi, formatting reference is uncrustify: https://docs.zephyrproject.org/latest/contribute/index.html#uncrustify.

Aim of these requests is to enhance and speed up review process and in the end global code base quality.

**Comment by erwango** [comment]
@cybertale

File: drivers/adc/adc_stm32.c [hunk 9]
```diff
@@ -187,7 +188,7 @@ struct adc_stm32_cfg {
 };
 
 static int check_buffer_size(const struct adc_sequence *sequence,
-			     u8_t active_channels)
+				 u8_t active_channels)
```
[the remark this answers is not in the document]
Comment (rickconrey) [reply drivers/adc/adc_stm32.c]: Thanks for the feedback.  I'll update this PR and keep this in mind for future contributions.

I used uncrustify and it updated the formatting, but it seems to clash with checkpatch.pl. I get a lot of "line over 80 character" warnings.  Am I doing something wrong with uncrustify? This is what I'm running:
uncrustify --replace --no-backup -l C -c $ZEPHYR_BASE/.uncrustify.cfg src.file

Also, I guess I don't have vim setup in the way that this project likes its formatting.  Is there a vimrc file I can follow to make my editor more friendly to zephyr?

File: drivers/adc/adc_stm32.c (hunk 2 above)
Comment (rickconrey) [reply drivers/adc/adc_stm32.c]: Pulled this one out into its own #if defined. I'm not sure if it's MCU specific, but I wasn't able to compile with the original code.

File: drivers/adc/adc_stm32.c (hunk 9 above)
[the remark this answers is not in the document]
Comment (erwango) [reply drivers/adc/adc_stm32.c]: Seems you are doing right.
I usually do a bit of clean up manually before running uncrustify. Maybe this is the reason why I didn't faced this before.
There is no .vimrc file available from what I know, but I'm sure you'll find some folks that may provide this on zephyr slack channel.

File: drivers/adc/adc_stm32.c (hunk 2 above)
Comment (cybertale) [reply drivers/adc/adc_stm32.c]: I see, the problem is I was using the legacy define of common adc instance pointer 'ADC', which should be deprecated and already commented as legacy in some SoCs' header files like stm32l083xx.h:782. There's no need to add another #if here, just change the 'ADC' in line 558 to __LL_ADC_COMMON_INSTANCE(adc) should be fine for all SoCs.

File: drivers/adc/adc_stm32.c [hunk 10]
```diff
@@ -267,6 +269,10 @@ static int start_read(struct device *dev, const struct adc_sequence *sequence)
 #endif
 	data->channel_count = 1;
 
+#if defined(CONFIG_SOC_SERIES_STM32WBX)
```
Comment (erwango) [inline drivers/adc/adc_stm32.c]: Out of curiosity, why is this required for STM32WB only (and not similar SoC L4 for instance)?
This would require a comment to make it clear

**Review by erwango — CHANGES_REQUESTED** [review]
Last comment.
Please also take into account @cybertale proposal.
Then, we'll be fine

File: drivers/adc/adc_stm32.c (hunk 10 above)
Comment (rickconrey) [reply drivers/adc/adc_stm32.c]: It looks like it isn't actually needed since the ADC defaults to single diff.  I'll remove it.

File: drivers/adc/adc_stm32.c [hunk 11]
```diff
@@ -585,6 +596,9 @@ static int adc_stm32_init(struct device *dev)
 	}
 #endif
 
+#ifdef CONFIG_SOC_SERIES_STM32WBX
+	LL_ADC_REG_SetTriggerSource(adc, LL_ADC_REG_TRIG_SOFTWARE);
```
Comment (erwango) [inline drivers/adc/adc_stm32.c]: Same question here (why STM32WB only).
Sorry, missed it previously.

**Review by erwango — CHANGES_REQUESTED** [review]
Very last comment.

File: drivers/adc/adc_stm32.c (hunk 11 above)
Comment (rickconrey) [reply drivers/adc/adc_stm32.c]: Same answer.  It's actually unnecessary since it is in software trigger mode by default.  I removed it.

**Review by dbkinder — APPROVED** [review]
doc changes LGTM.

**Review by erwango — APPROVED** [review]
